src/App.tsx

```diff
@@ -1,24 +1,147 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { TodoList } from './TodoList/TodoList';
+import {
+  applyUncompleted,
+  applyHasCompleted,
+  applySelectedTodos,
+  removeTodo,
+} from './helpers/helpers';
+import { Filter } from './types/Selected-filter-enum';
+import { ErrorType } from './types/errors-enum';
+import { Header } from './Header/Header';
+import { Footer } from './Footer/Footer';
+import { Notification } from './Notification/Notification';
 
-const USER_ID = 0;
+const USER_ID = 12057;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo []>([]);
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [filterType, setFilterType] = useState<Filter>(Filter.all);
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const hasTodosFromServer = todos.length !== 0;
+
+  const handleFilterType = (type: Filter) => {
+    setFilterType(type);
+  };
+
+  const handleTitle = (e:React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(e.target.value);
+  };
+
+  const handleError = (errorMessage: ErrorType) => {
+    setError(errorMessage);
+
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    setError(null);
+    getTodos(USER_ID)
+      .then((todosFS => setTodos(todosFS)))
+      .catch(() => handleError(ErrorType.didNotGetTodos));
+  }, []);
+
+  const onCreateTodo = () => {
+    const trimmedTitle = newTitle.trim();
+
+    setIsLoading(true);
+    setTempTodo({
+      id: 0,
+      title: trimmedTitle,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    createTodo({
+      title: trimmedTitle,
+      completed: false,
+      userId: USER_ID,
+    })
+      .then((todoFromServer) => {
+        setNewTitle('');
+        setTodos((currentTodos) => [...currentTodos, todoFromServer]);
+      })
+      .catch(() => {
+        handleError(ErrorType.cantUploadTodo);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setTempTodo(null);
+      });
+  };
+
+  const onDeleteTodo = (id: number) => {
+    deleteTodo(id)
+      .then(() => {
+        setTodos((currentTodos) => removeTodo(id, currentTodos));
+      })
+      .catch(() => {
+        handleError(ErrorType.cantDeleteTodo);
+      });
```

Comment:
You should show loaders for all todos that processing now, for now, I don't see any loaders when deleting one todo or click on the clear completed button
You can create a state with todos ids that currently updating and depends on it show loaders.
Simple example:
```typescript
const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([]);
```

```suggestion
    setLoadingTodosIds(prev => [...prev, id]);
    
    deleteTodo(id)
      .then(() => {
        setTodos((currentTodos) => removeTodo(id, currentTodos));
      })
      .catch(() => {
        handleError(ErrorType.cantDeleteTodo);
      })
      .finally(() => {
        setLoadingTodosIds(prev => prev.filter(todoId => todoId !== id));
      });
```